src/glewlwyd.c

```diff
@@ -952,11 +959,43 @@ int build_config_from_file(struct config_elements * config) {
   
   auth = config_setting_get_member(root, "authentication");
   if (auth != NULL) {
+    config_setting_lookup_bool(auth, "http_auth", &cur_http_auth);
+    config->has_auth_http = cur_http_auth;
+    if (config->has_auth_http &&
+        config->use_scope) {
+      config_destroy(&cfg);
+      fprintf(stderr, "Error, due to security concerns you can not use authentication via HTTP together with scopes\n");
+      return 0;
+    }
     config_setting_lookup_bool(auth, "database_auth", &cur_database_auth);
     config->has_auth_database = cur_database_auth;
     config_setting_lookup_bool(auth, "ldap_auth", &cur_ldap_auth);
     config->has_auth_ldap = cur_ldap_auth;
-    
+
+    if (config->has_auth_http) {
+      config_setting_lookup_string(auth, "http_auth_url", &cur_auth_http_auth_url);
+      if (cur_auth_http_auth_url != NULL) {
+        config->auth_http = o_malloc(sizeof(struct _auth_http));
+        if (config->auth_http == NULL) {
+          config_destroy(&cfg);
+          fprintf(stderr, "Error allocating resources for config->auth_http\n");
+          return 0;
+        } else {
+          memset(config->auth_http, 0, sizeof(struct _auth_http));
```

Comment:
In my first version I had two pointers in the struct. In case you have a self signed certificate on your server, I wanted to also give the certificat chain to check it. But in the end it was easier to use letsencrypt than to let curl do the check :-).

Reply:
Actually, I like this better, since there is the option `struct _u_request.check_server_certificate` to force checking the server certificate or not.
Especially in an embedded environment, it would be better to make it possible for the programmer to enable or not the server certificate checking.